### DISCUSSION TITLE
[typescript-react-query] How to import useQuery from my own instead react-query

### DISCUSSION THREAD

**Discussion by vctqs1** [dotansimha/graphql-code-generator#9521] [General]
Hi everyone. the generated file
```
import { useQuery } from 'react-query';

import { eurekaGraphqlRequest } from '../../../internals/hasura-client/execute-query';
import type * as Types from '../root-types';

import type { UseQueryOptions } from 'react-query';
export type StudyPlanItemAttrsFragment = { study_plan_item_id: string, available_from?: any | null, available_to?: any | null, content_structure?: any | null, start_date?: any | null, end_date?: any | null, status?: string | null, assignment_study_plan_item?: { assignment_id: string } | null, lo_study_plan_item?: { lo_id: string } | null };

```

with useQuery from `react-query`
but i'd like do some customize on useQuery and use from my own instead react-query

```
import { useQuery } from './my-config';

```

I tried to add config like this [EX](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-react-query#using-fetch-with-codegen-configuration)
```
plugins: [{
                        add: {
                            content: "import { useQuery } from './my-config';"
                        }
                    }
```

But instead replace the import, it just add `one-more-import-line`
![image](https://github.com/dotansimha/graphql-code-generator/assets/30227910/64cf6d4c-e03a-491c-a36b-9fc7fb270851)

Do we have any way to adjust this?

**Comment by vctqs1** [comment]
PR https://github.com/dotansimha/graphql-code-generator-community/pull/369
Issue https://github.com/dotansimha/graphql-code-generator-community/issues/368

**Comment by khitrenovich** [comment]
It would be great to have more generic solution - I just run into the same need for `typescript-react-apollo`...

**Comment by vctqs1** [comment]
@khitrenovich the PR already merge, hope it help your case
